nxcomp/src/Loop.cpp

```diff
@@ -6535,12 +6564,18 @@ int PrepareProxyConnectionUnix(char** path, int* timeout, int* proxyFileDescript
 
   /* FIXME: Add socket file existence and permission checks */
 
+
   *proxyFileDescriptor = -1;
   *reason = -1;
 
+  // determine the maximum number of characters that fit into struct
+  // sockaddr_un's sun_path member
+  const int sockpathlen =
+    sizeof(struct sockaddr_un) - offsetof(sockaddr_un, sun_path);
+
   sockaddr_un addr;
   addr.sun_family = AF_UNIX;
-  strncpy(addr.sun_path, *path, 108 - 1);
+  strncpy(addr.sun_path, *path, sockpathlen - 1);
```

Comment:
Should actually use `snprintf` here, since `strncpy` is not NULL-terminating the string in the worst case.

Reply:
have replaced almost all strncpys in nxcomp now